File: CS/WebApplication1/CustomDrillThroughProcessor.cs
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Web.WebDocumentViewer;
using WebApplication1.Reports;

namespace WebApplication1 {
    public class CustomDrillThroughProcessor : IWebDocumentViewerDrillThroughProcessor {
        public XtraReport CreateReport(DrillThroughContext context) {
            if (context.CustomData == "#back")
                return new MainReport();
            if (context.CustomData == "#detail1")
                return new DetailReport1();
            if (context.CustomData == "#detail2")
                return new DetailReport2();
            return context.Report;
        }
    }
}

File: VB/WebApplication1/CustomDrillThroughProcessor.vb
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Web.WebDocumentViewer
Imports WebApplication1.Reports

Namespace WebApplication1

    Public Class CustomDrillThroughProcessor
        Implements IWebDocumentViewerDrillThroughProcessor

        Public Function CreateReport(ByVal context As DrillThroughContext) As XtraReport Implements IWebDocumentViewerDrillThroughProcessor.CreateReport
            If Equals(context.CustomData, "#back") Then Return New MainReport()
            If Equals(context.CustomData, "#detail1") Then Return New DetailReport1()
            If Equals(context.CustomData, "#detail2") Then Return New DetailReport2()
            Return context.Report
        End Function
    End Class
End Namespace
